adobe/react-spectrum
# fix(#2901): ColorArea followup

1. aria-label for ColorArea inputs on desktop are simply "Color picker."
2. externally set aria-label is prepended to the default accessibility name so that "Color picker" is always included as part of the accessibility name.
3. On first focus, when interacting with the mouse, and on blur, inputs will have aria-valuetext set to channel name and value for both channels.
4. After a keyboard interaction, aria-valuetext will be the channel name and value for the channel that changed.
~5. We still use title to provide all three channel names and values as help text/accessibility description. When we include a color name string, it should be prepended to this.~
5. Remove use of `title` attribute to announce the full RGB/HSL/HSB value. Deferring until we have an API for providing color name.
6. Remove x/y translation string, because it is no longer needed.
7. Fix z-index of ColorLoupe
8. So that only a single "2d slider" control shows up when listing form elements for screen readers, add `aria-hidden="true"` to the unfocused input when the value has not changed via the keyboard, but remove `aria-hidden` to reveal the input for each channel when the value has changed with the keyboard.


Closes #2901 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue: #2901.

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/5a6c7cb40a9fa380a8618d27d156d9b49da6b9b1/storybook/index.html?path=/story/colorarea--x-blue-y-green and move focus to the Storybook story iframe.
2. Turn on VoiceOver
3. Press Control+Option+U (possibly followed by ArrowRight) to bring up a list of Form Controls in the Storybook story.
4. The first Form Control in the list should be identified as "Blue: 255, Green: 255, foo Color picker, 2d slider"
5. Press ArrowDown to highlight the "Blue: 255, Green: 255, foo Color picker, 2d slider" .
6. Press Enter or Space key to set focus on the "Blue: 255, Green: 255, foo Color picker, 2d slider".
7. Use ArrowLeft to decrease the value for the x-Channel, Blue by 1.
8. VoiceOver will announce the value change as "Blue: 254"
9. Press fn+ArrowDown to trigger a PageDown event, decreasing the value for the y-Channel, Green by its page step, 17.
10. VoiceOver will announce the value change as "Green: 238, foo Color picker, 2d Slider." Note that whenever you change the value along a different axis, the screen reader will announce the accessibility name and role description for the input after the event, because the focus has changed.
11. Use ArrowUp to increase the value for the y-Channel, Green by 1. 
12. VoiceOver will announce the value change as "Green: 239."
13. Press Control+Option+U  to bring up the list of Form Controls in the Storybook story again.
14. The first two Form Controls in the list should be identified as "Blue: 254, foo, Color picker, 2d slider" and "Green 238, foo, Color picker, 2d slider". We show both inputs after a value change using the keyboard, so that both values will be displayed in the list of controls.
15. Press ArrowDown to highlight the "Blue: 254, foo, Color picker, 2d slider" .
16. Press Enter or Space key to set focus on the "Blue: 254, Green: 239, foo, Color picker, 2d slider". Note that VoiceOver announces both the x and y channel names and values, because the focus has now moved from the y-Channel input to the x-Channel input. When you select and restore focus to the same channel input that you just changed using the keyboard, only the channel name and value for that input will be announced.
17. Change the value again using the arrow keys.
18. Use mouse to click on the ColorArea gradient to change the value. Notice that when changing the value using the mouse, VoiceOver will always announces both the x and y channel names and value. 
19. Change the value again using the arrow keys, so VoiceOver announces only the channel name and value for the axis that changed.
20. Press Tab to move focus to the Red channel ColorSlider.
21. Press Shift+Tab to move focus back to the ColorArea.
22. Notice that when restoring focus to the ColorArea VoiceOver again announces both the x and y channel names and values.

## 🧢 Your Project:

Adobe/Accessibility
